ugvc/filtering/transformers.py

```diff
@@ -0,0 +1,329 @@
+
+    def ins_del_encode_df(df):
+        return pd.DataFrame(np.array(df[0].apply(ins_del_encode)).reshape(-1, 1), index=df.index)
+
+    default_filler = impute.SimpleImputer(strategy="constant", fill_value=0)
+    tuple_filter = preprocessing.FunctionTransformer(tuple_encode_df)
+    ins_del_encode_filter = preprocessing.FunctionTransformer(ins_del_encode_df)
+    tuple_encode_df_transformer = preprocessing.FunctionTransformer(tuple_encode_df)
+    tuple_encode_doublet_df_transformer = preprocessing.FunctionTransformer(tuple_encode_doublet_df)
+    tuple_uniform_encode_df_transformer = preprocessing.FunctionTransformer(tuple_uniform_encode)
+    left_motif_filter = preprocessing.FunctionTransformer(motif_encode_left_df)
+    right_motif_filter = preprocessing.FunctionTransformer(motif_encode_right_df)
+    allele_filter_single = preprocessing.FunctionTransformer(allele_encode_single)
+    allele_filter = preprocessing.FunctionTransformer(allele_encode_df)
+    gt_filter = preprocessing.FunctionTransformer(gt_encode_df)
+
+    transform_list = [
+        ("sor", default_filler, ["sor"]),
+        ("dp", default_filler, ["dp"]),
+        ("alleles", allele_filter, "alleles"),
+        ("x_hin", make_pipeline(tuple_filter, allele_filter_single), "x_hin"),
+        ("x_hil", make_pipeline(tuple_filter, default_filler), "x_hil"),
+        ("x_il", make_pipeline(tuple_filter, default_filler), "x_il"),
+        ("indel", "passthrough", ["indel"]),
+        ("x_ic", make_pipeline(tuple_filter, ins_del_encode_filter), "x_ic"),
+        ("x_lm", left_motif_filter, "x_lm"),
+        ("x_rm", right_motif_filter, "x_rm"),
+        ("x_css", make_pipeline(tuple_filter, preprocessing.OrdinalEncoder()), "x_css"),
+        ("x_gcc", default_filler, ["x_gcc"]),
+    ]
+    features = [x[0] for x in transform_list]
+
+    if vtype == VcfType.DEEP_VARIANT:
+        transform_list.extend(
+            [
+                ("vaf", tuple_filter, "vaf"),
+                ("ad", tuple_encode_doublet_df_transformer, "ad"),
+                ("mq0_ref", default_filler, ["mq0_ref"]),
+                ("mq0_alt", default_filler, ["mq0_alt"]),
+                ("ls_ref", default_filler, ["ls_ref"]),
+                ("ls_alt", default_filler, ["ls_alt"]),
+                ("rs_ref", default_filler, ["rs_ref"]),
+                ("rs_alt", default_filler, ["rs_alt"]),
+                ("mean_nm_ref", default_filler, ["mean_nm_ref"]),
+                ("median_nm_ref", default_filler, ["median_nm_ref"]),
+                ("mean_nm_alt", default_filler, ["mean_nm_alt"]),
+                ("median_nm_alt", default_filler, ["median_nm_alt"]),
+                ("mean_mis_ref", default_filler, ["mean_mis_ref"]),
+                ("median_mis_ref", default_filler, ["median_mis_ref"]),
+                ("mean_mis_alt", default_filler, ["mean_mis_alt"]),
+                ("median_mis_alt", default_filler, ["median_mis_alt"]),
```

Comment:
where are all of these fields come from?

Reply:
From my tests of filtering FFPE variants. these were added to the VCF/Dataframe by custom code. 
Currently I feel like I don't have a very good mechanism of supplying custom transformers